src/test/java/org/embulk/input/zendesk/services/TestZendeskSupportAPIService.java

```diff
@@ -17,9 +22,11 @@
 import static org.mockito.ArgumentMatchers.anyBoolean;
 import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.times;
 import static org.mockito.Mockito.verify;
 import static org.mockito.Mockito.when;
 
+import java.time.Instant;
 import java.util.Collections;
 
 public class TestZendeskSupportAPIService
```

Comment:
This test class contains many test cases which are in abstract parent class `ZendeskNormalService`, I'd suggest we create another test class for that abstract one to keep it isolated with the subclasses. Subclasses only test what happen in those subclasses.

It will give us the advantage if we implement new services in that future which are similar as support, we just need to add test cases for those services only. Or if we need to update `ZendeskNormalService` we only need to update test cases in that abstract one instead of update other subclasses test cases.

Reply:
fixed